File: examples/loops/fibonacci.cpp
#include <cstdlib>
#include <iostream>
#include <vector>

int main() {

    std::vector<long> fib{0, 1};

    std::cout << "How many terms of the Fibonacci sequence to compute? (N > 2) ";

    int N{};
    std::cin >> N;

    if (N <= 2) {
        // we already have that -- abort
        std::cout << "too few" << std::endl;
        std::exit(1);
    }

    // add to the sequence

    while (static_cast<int>(fib.size()) < N) {
        auto s = fib.size();
        fib.push_back(fib[s-1] + fib[s-2]);
    }

    // now output

    for (auto e : fib) {
        std::cout << e << " ";
    }
    std::cout << std::endl;

}

File: examples/python/fibonacci.py
import sys

# initialize a list with 2 elements

fib = [0, 1]

N = int(input("How many terms of the Fibonacci sequence to compute? (N > 2) "))

if N <= 2:
    # we've already got 2
    sys.exit("Error: must enter N > 2")

# add to the sequence
while len(fib) < N:
    fib.append(fib[-1] + fib[-2])

# now output -- we can just print the list
print(fib)

